lighthouse-core/lib/i18n/i18n.js

```diff
@@ -238,14 +248,23 @@ function getRendererFormattedStrings(locale) {
 }
 
 /**
+ * Register a file's UIStrings with i18n, return function to
+ * generate the string ids.
+ *
  * @param {string} filename
  * @param {Record<string, string>} fileStrings
  */
 function createMessageInstanceIdFn(filename, fileStrings) {
   /** @type {Record<string, string>} */
   const mergedStrings = {...UIStrings, ...fileStrings};
 
-  /** @param {string} icuMessage @param {*} [values] */
+  /**
+   * Convert a message string & replacement values into an
+   * indexed id value in the form '{messageid} | # {index}'.
+   *
+   * @param {string} icuMessage
+   * @param {*} [values]
```

Comment:
at some point we decided this could be `Record<string, string | number>`, but that idea didn't spread. We probably should, though.

Reply:
added as a follow up.